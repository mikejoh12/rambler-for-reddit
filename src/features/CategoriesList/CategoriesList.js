import React, { useEffect } from 'react'
import { useSelector } from 'react-redux'
import { selectCategories, fetchPosts } from '../reddit/redditSlice'
import { List, ListItem, ListItemText } from '@material-ui/core'
import { useDispatch } from 'react-redux'
import { currentTopicUpdated } from '../reddit/redditSlice'


const CategoriesListItem = ({ category }) => {
    const dispatch = useDispatch()

    const onCategoryChanged = () => {
        dispatch(currentTopicUpdated(category))
    }

    return <ListItem  
                    button
                    onClick={onCategoryChanged}>
                <ListItemText primary={category} />
            </ListItem>
}

export const CategoriesList = () => {
    const categories = useSelector(selectCategories);
    const dispatch = useDispatch()

    //Fetch new posts
    const currentTopic = useSelector(state => state.reddit.currentTopic)
    useEffect(() => {
        dispatch(fetchPosts(currentTopic))
    }, [currentTopic, dispatch])

    const categoriesList = categories.map(category => {
        return <CategoriesListItem 
            category={category}
            key={category} />
    })

    return (
        <List>
            {categoriesList}
        </List>
    )
}