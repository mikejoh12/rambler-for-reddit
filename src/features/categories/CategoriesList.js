import React from 'react'
import { useSelector } from 'react-redux'
import { selectCategories } from '../reddit/redditSlice'
import { Link } from 'react-router-dom'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardActionArea from '@material-ui/core/CardActionArea'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'
//import { Grid } from '@material-ui/core'

const CategoriesListItem = ({ category, icon_img }) => {

    const useStyles = makeStyles({
        root: {
          margin: 10,
          backgroundColor: '#D3E3F0',
          color: '#181EE3',
        }
      });
 
    const classes = useStyles()

    const handleClick = () => {
        window.scrollTo(0, 0)
    }

    return (
        <Card className={classes.root}>
            <CardActionArea component={Link} to={`/r/${category}`} onClick={handleClick}>
            <CardContent>
                <img src={icon_img} height="25" alt=""></img>
                <Typography variant="body1">
                    {category}
                </Typography>
            </CardContent>
            </CardActionArea>
        </Card>
    )
}

export const CategoriesList = () => {
    const categories = useSelector(selectCategories);

    return (
            <div>
                <Typography variant="h5" color="textPrimary">
                    Subreddits
                </Typography>
                    {
                        categories && categories.map(item => {
                            return <CategoriesListItem 
                                category={item.subreddit}
                                key={item.subreddit}
                                icon_img={item.icon_img} />
                        })
                    }
            </div>
    )
}