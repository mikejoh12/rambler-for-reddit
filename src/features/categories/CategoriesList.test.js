import { CategoriesList } from './CategoriesList'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
const { shallow } = require("enzyme")

const mockStore = configureStore([])
const store = mockStore()

describe('CategoriesList component', () => {

    it('should render with given state from Redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <CategoriesList />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})