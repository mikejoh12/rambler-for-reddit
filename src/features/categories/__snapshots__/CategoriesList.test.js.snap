// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesList component should render with given state from Redux store 1`] = `ShallowWrapper {}`;
